content/en/blog/_posts/2023-12-13-kubernetes-1.29.md

```diff
@@ -0,0 +1,243 @@
+
+### Node volume expansion Secret support for CSI drivers ([SIG Storage](https://github.com/kubernetes/community/tree/master/sig-storage)) {#csi-node-volume-expansion-secrets}
+    
+In Kubernetes, a volume expansion operation may include the expansion of the volume on the node, which involves filesystem resize. Some CSI drivers require secrets, for example a credential for accessing a SAN fabric, during the node expansion for the following use cases:
+* When a PersistentVolume represents encrypted block storage, for example using LUKS, you may need to provide a passphrase in order to expand the device.
+* For various validations, the CSI driver needs to have credentials to communicate with the backend storage system at time of node expansion.
+
+To meet this requirement, the CSI Node Expand Secret feature was introduced in Kubernetes v1.25. This allows an optional secret field to be sent as part of the NodeExpandVolumeRequest by the CSI drivers so that node volume expansion operation can be performed with the underlying storage system. In Kubernetes v1.29, this feature became generally available.
+
+### KMS v2 encryption at rest generally available ([SIG Auth](https://github.com/kubernetes/community/tree/master/sig-auth)) {#kms-v2-api-encryption}
+ 
+One of the first things to consider when securing a Kubernetes cluster is encrypting persisted
+API data at rest. KMS provides an interface for a provider to utilize a key stored in an external
+key service to perform this encryption. With the Kubernetes v1.29, KMS v2 has become
+a stable feature bringing numerous improvements in performance, key rotation,
+health check & status, and observability.
+These enhancements provide users with a reliable solution to encrypt all resources in their Kubernetes clusters. You can read more about this in [KEP-3299](https://kep.k8s.io/3299).
+
+For Kubernetes v1.29, [encryption at rest for API data](/docs/tasks/administer-cluster/encrypt-data/)
+can integrate with an external _key management service_ (KMS). Available since Kubernetes v1.13,
+this mechanism allows you to protect your cluster's control plane from security risks even if the etcd
+persistence layer is somehow compromised.
+
+New in v1.29 is stable support for version 2 of the KMS integration. If you're using KMS for
+encryption, or you plan to, the Kubernetes project recommends using the version 2 integration
+mechanism in all cases.
+
+For details about integration with specific cloud providers and other services, ask your provider
+or consult the documentation for their plugin.
+
+## Improvements that graduated to beta in Kubernetes v1.29 {#graduations-to-beta}
+
+_This is a selection of some of the improvements that are now beta following the v1.29 release._
+
+### Per-plugin callback functions in kube-scheduler for accurate requeueing ([SIG Scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling)) {#scheduler-queueing-hint}
```

Comment:
It's enabled by default in 1.28.0 - 1.28.5, but is disabled by default in 1.29 because we found it consuming memory a lot. 

FYI: 
- https://github.com/kubernetes/website/pull/44316
- https://kubernetes.slack.com/archives/C2C40FMNF/p1702438650646789